DiverseSelector/diversity.py

```diff
@@ -96,53 +100,46 @@ def entropy(x: np.ndarray) -> float:
     Returns
     -------
     e : float
-        Entropy of matrix.
+        Entropy of matrix in the range [0,1].
 
     Notes
     -----
-    Feature matrices are converted to bits,
-    so we lose any information associated with num in matrix.
+    The feature matrix does not need to be in bitstring form to use this function.
+    However, the features should be encoded in a binary way, such that 0 means
+    absence of the feature, and nonzero means presence of the feature.
+
     Weidlich, I. E., and Filippov, I. V. (2016)
     Using the Gini coefficient to measure the chemical diversity of small-molecule libraries.
     Journal of Computational Chemistry 37, 2091-2097.
     """
 
-    # convert input matrix to bit matrix
-    y = np.empty(x.shape)
-    for i in range(0, len(x)):
-        for j in range(0, len(x[0])):
-            if x[i, j] != 0:
-                y[i, j] = 1
-            else:
-                y[i, j] = 0
     # initialize variables
     length = len(x[0])
     n = len(x)
```

Comment:
These two variables seem to represent the same value. Any reason we need both?

Reply:
They are not the same value, n is the number of molecules and length is the number of features. I've renamed them in my latest commit to make this clearer.

Reply:
Yup, you are right. they are the number of rows and columns. It is a strange way of getting rows and columns, but I will fix it myself.